server/auth_test.go

```diff
@@ -229,3 +219,147 @@ func TestSwagger(t *testing.T) {
 		})
 	}
 }
+
+func TestPermissionsForSTTChecksPage(t *testing.T) {
+	ts := strconv.FormatInt(time.Now().Unix(), 10)
+	none := "none-" + ts
+	viewer := "viewer-" + ts
+	editor := "editor-" + ts
+	admin := "admin-" + ts
+
+	noneID := createUser(t, none)
+	defer deleteUser(t, noneID)
+
+	viewerID := createUserWithRole(t, viewer, "Viewer")
+	defer deleteUser(t, viewerID)
+
+	editorID := createUserWithRole(t, editor, "Editor")
+	defer deleteUser(t, editorID)
+
+	adminID := createUserWithRole(t, admin, "Admin")
+	defer deleteUser(t, adminID)
+
+	tests := []struct {
+		name       string
+		login      string
+		statusCode int
+	}{
+		{name: "default", login: none, statusCode: http.StatusUnauthorized},
+		{name: "viewer", login: viewer, statusCode: http.StatusUnauthorized},
+		{name: "editor", login: editor, statusCode: http.StatusUnauthorized},
+		{name: "admin", login: admin, statusCode: http.StatusOK},
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run("get settings/"+test.name, func(t *testing.T) {
```

Comment:
looks like we have the same code in both tests except path, maybe we can extract it to an array and just loop over it?

Reply:
Done